operator/pkg/apis/cassandra/v1beta1/seed_providers.go

```diff
@@ -0,0 +1,151 @@
+// Copyright DataStax, Inc.
+// Please see the included license file for details.
+
+package v1beta1
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/datastax/cass-operator/operator/pkg/utils"
+)
+
+type SeedProvider struct {
+	ClassName   string
+	Seeds       []string
+	OtherParams []map[string]interface{}
+}
+
+func buildClassNameForSeedProvider(serverType string) string {
+	var className string
+	if serverType == "cassandra" {
+		className = "org.apache.cassandra.locator.K8SeedProvider"
+	} else {
+		className = "org.apache.cassandra.locator.SimpleSeedProvider"
+	}
+	return className
+}
+
+func (dc *CassandraDatacenter) buildDefaultSeedProvider() *SeedProvider {
+	return &SeedProvider{
+		ClassName:   buildClassNameForSeedProvider(dc.Spec.ServerType),
+		Seeds:       dc.Spec.StaticSeedIPs,
+		OtherParams: nil,
+	}
+}
+
+func findSeedsAndOtherParams(parameters interface{}) ([]map[string]interface{}, []string) {
+	var otherParams []map[string]interface{}
+	var seeds []string
+
+	switch parameters := parameters.(type) {
+	case []map[string]interface{}:
+		for _, m := range parameters {
+			if v, ok := m["seeds"]; ok {
+				switch v := v.(type) {
+				case []string:
+					seeds = v
+				}
+			} else {
+				otherParams = append(otherParams, m)
+			}
+		}
+	}
+
+	return otherParams, seeds
+}
+
+func (dc *CassandraDatacenter) getSeedProviderData(sp map[string]interface{}) *SeedProvider {
+	// no existing seed providers found in dc spec config
+	if sp == nil || len(sp) < 1 {
+		return nil
+	}
+
+	parameters := sp["parameters"]
+	//no parameters found in seed provider
+	if parameters == nil {
+		return nil
+	}
+
+	otherParams, seeds := findSeedsAndOtherParams(parameters)
+
+	var className string
+	if c, ok := sp["class_name"]; ok {
+		className = fmt.Sprintf("%v", c)
+	} else {
+		className = buildClassNameForSeedProvider(dc.Spec.ServerType)
+	}
+
+	return &SeedProvider{ClassName: className, Seeds: seeds, OtherParams: otherParams}
```

Comment:
It doesn't look like we really care about the seed provider here. I think you just want to use the `seed` option we give to config builder as part of the `cluster-info`. See:

https://github.com/datastax/cass-operator/blob/master/operator/pkg/serverconfig/configgen.go#L14-L28

and

https://github.com/datastax/cass-operator/blob/248b30efe797d0656f2fc5c8e96dc3c431ab9a32/operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go#L432-L435

This way we are letting config builder handle the details of figuring out the correct class name given the cassandra version.